staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -545,9 +545,13 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 		reqScope.MetaGroupVersion = *a.group.MetaGroupVersion
 	}
 	for _, action := range actions {
-		versionedObject := storageMeta.ProducesObject(action.Verb)
-		if versionedObject == nil {
-			versionedObject = defaultVersionedObject
+		producedObject := storageMeta.ProducesObject(action.Verb)
+		if producedObject == nil {
+			producedObject = defaultVersionedObject
+		}
+		connectProducedObject := storageMeta.ProducesObject(action.Verb)
```

Comment:
Shouldn't this be in the ConnectMethods() loop?

Reply:
Yes, sorry. Amended and repushed.